### PR TITLE
Migrate ModelInterface functionality to parse actions

### PR DESCRIPTION
**Pull request by jonc125**
This PR contains enough to get the GraphState protocol working again without units conversions.

It also updates a few things to the latest cellmlmanip.

### CODE REVIEW

File: fc/code_generation.py [hunk 1]
```diff
@@ -220,7 +218,7 @@ def derivative_name(deriv):
 
         # Allow special output value 'state_variable'
         # TODO Replace this with a more generic implementation
-        if output == 'state_variable':
+        if output[1] == 'state_variable':
```
Comment (MichaelClerx) [inline fc/code_generation.py]: I didn't realise `state_variable` had a URI?

File: fc/parsing/actions.py [hunk 2]
```diff
@@ -483,46 +488,55 @@ def _expr(self):
...
-    # def _xml(self):
-    #     return P.setIndependentVariableUnits(**self.get_attribute_dict('units'))
-    pass
+    """Parse action for specifying the units time should have in the model.
+
+    ``independent var units <uname>``
+    """
+    def _expr(self):
+        self.time_units = self.get_named_token_as_string('units')
+        assert self.time_units is not None
```
Comment (MichaelClerx) [inline fc/parsing/actions.py]: Is this a sanity check, or can this show up in the user's error output?

File: fc/parsing/actions.py [hunk 3]
```diff
@@ -610,20 +636,58 @@ class UnitsConversion(BaseGroupAction):
...
-            InputVariable,
-            ModelEquation,
-        )
-        for action in self:
-            if isinstance(action, handled):
-                output.append(action.expr())
-            # TODO: Create objects for all parts of the model interface
-        return output
+        actions = self.get_children_expr()
+        print('in MI._expr', self.tokens, actions, map(type, actions))
```
Comment (MichaelClerx) [inline fc/parsing/actions.py]: print!

File: fc/code_generation.py (hunk 1 above)
Comment (jonc125) [reply fc/code_generation.py]: Yep, it's a standard oxmeta:state_variable reference.

File: fc/parsing/actions.py (hunk 2 above)
Comment (jonc125) [reply fc/parsing/actions.py]: Should be just a sanity check - the parser would complain before now if it didn't exist.

File: fc/parsing/actions.py (hunk 3 above)
Comment (jonc125) [reply fc/parsing/actions.py]: Oops!

**Pull request merged**